www/app/search/search.controller.spec.js

```diff
@@ -0,0 +1,55 @@
+/* jshint -W117, -W030 */
+describe('SearchController', function() {
+  var controller,
+      deferredSearch,
+      deferredToTablePage,
+      tablesServiceMock,
+      yelpServiceMock;
+
+  // Load the App Module to access controllers
+  beforeEach(module('app'));
+
+  //stop caching for HTML load error
+  beforeEach(module(function($provide, $urlRouterProvider) {
+    $provide.value('$ionicTemplateCache', function() {});
+    $urlRouterProvider.deferIntercept();
+  }));
+
+  // instantiate the controller and mocks for every test
+  beforeEach(inject(function($controller, $q) {
+    deferredSearch = $q.defer();
+    deferredToTablePage = $q.defer();
+
+    // mock services
+    tablesServiceMock = {
+      toTablePage: jasmine.createSpy('toTablePage spy').and.returnValue(deferredToTablePage.promise)
+    };
+
+    yelpServiceMock = {
+      yelpSearch: jasmine.createSpy('yelpSearch spy').and.returnValue(deferredSearch.promise),
+    };
+
+    // instantiate SearchController
+    controller = $controller('SearchController', {
+      'yelpService': yelpServiceMock,
+      'tablesService': tablesServiceMock
+    });
+  }));
+
+  // call doLogin on the controller for every test
+  beforeEach(inject(function(_$rootScope_) {
+    $rootScope = _$rootScope_;
+    controller.restSearch('rest', 'loc');
+    controller.tablePage('id');
+  }));
+
```

Comment:
I believe all of these beforeEach functions can be combined into one, unless you only want one to run for the #SearchController, and if so, then it needs to be put in that describe block.